raco/backends/myria/myria.py

```diff
@@ -1268,7 +1268,16 @@ def fire(self, exp):
             return MyriaLimit(exp.count,
                               algebra.Collect(
                                   MyriaLimit(exp.count, exp.input)))
+        return exp
+
+
+class GlobalOrderBy(rules.Rule):
 
+    def fire(self, exp):
+        if exp.__class__ == algebra.OrderBy:
+            return MyriaInMemoryOrderBy(
```

Comment:
You may actually need a special rule that recognizes just the `Limit(OrderBy)` pattern (there's no point in having a rule `OrderBy -> OrderBy(Collect(OrderBy))`, although if we had a `Merge` exchange operator it would be sensible to have a rule `OrderBy -> Merge(OrderBy)`). Also the `CollectBeforeLimit` rule may need to be modified to ignore this pattern (I think it's really useless anyway since there's not much point in using `LIMIT` without `ORDER BY`).